compiler/sem/mirconstr.nim

```diff
@@ -0,0 +1,72 @@
+## This module contains constructor procedures for the different kinds of
+## ``MirNode``s plus convenience procedures for generating expressions.
+
+import
+  compiler/ast/ast_types,
+  compiler/sem/mirtrees
```

Comment:
any particular reason to separate these out from `mirtrees`?

Reply:
The construction procedures and utilities are only needed when generating new MIR code. For just comprehending and processing MIR (e.g. `mirtraverse.nim`, `astgen.nim`, etc.), they're not needed, so I thought it'd make sense to put them into a separate module.

Reply:
Fair enough, I can go either way on this sort of thing. Maaaaybe consider renaming it `mirtrees_constr.nim` so it feels a bit subordinate to the other module. Might hint at "don't read me first".